feat: update onboarding steps to include sharing the DevCard

brandonroberts opened the pull request:
## Description

<!-- 
Please do not leave this blank 
This PR [adds/removes/fixes/replaces] the [feature/bug/etc]. 
-->
Updates the onboarding flow to authenticate, set your time zone, and share your dev card.

## Related Tickets & Documents
<!-- 
Please use this format link issue numbers: Fixes #123
https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword 
-->

## Mobile & Desktop Screenshots/Recordings

<!-- Visual changes require screenshots -->


https://github.com/user-attachments/assets/042748e6-c594-43b2-a5c3-8bbea4110fb9



## Steps to QA
<!-- 
Please provide some steps for the reviewer to test your change. If you have wrote tests, you can mention that here instead.

1. Click a link
2. Do this thing
3. Validate you see the thing working
-->
1. Reset your onboarding flag
2. Login
3. Click on `Complete Onboarding` from the nav bar
4. Go through onboarding

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [x] Tier 4

## [optional] What gif best describes this PR or how it makes you feel?

<img src="https://media1.giphy.com/media/A1oBMukTqFfkoY1HiH/giphy.gif"/>

<!-- note: PRs with deleted sections will be marked invalid -->

<!--
  For Work In Progress Pull Requests, please use the Draft PR feature,
  see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.
  
  For a timely review/response, please avoid force-pushing additional
  commits if your PR already received reviews or comments.
  
  Before submitting a Pull Request, please ensure you've done the following:
  - 📖 Read the Open Sauced Contributing Guide: https://github.com/open-sauced/.github/blob/main/CONTRIBUTING.md.
  - 📖 Read the Open Sauced Code of Conduct: https://github.com/open-sauced/.github/blob/main/CODE_OF_CONDUCT.md.
  - 👷‍♀️ Create small PRs. In most cases, this will be possible.
  - ✅ Provide tests for your changes.
  - 📝 Use descriptive commit messages.
  - 📗 Update any related documentation and include any relevant screenshots.
-->

nickytonline commented on pages/start.tsx:
```diff
@@ -242,12 +166,12 @@ const LoginStep3: React.FC<LoginStep3Props> = ({ interests, user }) => {
           Authorization: `Bearer ${sessionToken}`,
         },
-        body: JSON.stringify({ interests, timezone }),
+        body: JSON.stringify({ interests: [], timezone }),
```
this is because we replace interests now with go to dev card but still need a value set for interests?

brandonroberts replied:
Yes, its still required in the API for now

nickytonline replied:
I created https://github.com/open-sauced/api/issues/1030 for this.

nickytonline commented on pages/start.tsx:
```diff
@@ -70,7 +68,7 @@ const LoginStep1: React.FC<LoginStep1Props> = ({ user }) => {
       setQueryParams({ step: "2" } satisfies QueryParams);
     }
-  }, [user, onboarded, providerToken]);
+  }, [user, onboarded, providerToken, router]);
```
did this get added because of a linter warning?

brandonroberts replied:
Yes, I can take it out though. I thought it was an error

nickytonline replied:
It's because it's used in the useEffect. Was just checking is all. Feel free to keep it in.

nickytonline reviewed (approved):
Works great on mobile and desktop. 🚢

isabensusan reviewed (approved):
I'd say it's good to ship as is! 

But if possible, is there any chance we can add a preview of the card in this step? Even the OG image could work. Or as a second resource, a random example image?

(very rough mock!! but you get the idea)

<img width="1162" alt="image" src="https://github.com/user-attachments/assets/00d891fc-4c06-420c-98ea-13b63f940b0d">

BekahHW reviewed (changes requested):
My concern here is that we don't have a clear path for what they should do after they share their devcard and we give them no other options in the onboarding path. Can we give an option? Share your DevCard or Get Started Now! The latter links to the Workspace. 

It would also be nice if once a person shares a DevCard during onboarding, they're taken to the profile edit page. 

We don't want to leave them hanging after they share. 

I'm not trying to nitpick, but I want to make sure users can see a clear path to usage.

brandonroberts commented:
> My concern here is that we don't have a clear path for what they should do after they share their devcard and we give them no other options in the onboarding path. Can we give an option? Share your DevCard or Get Started Now! The latter links to the Workspace.
> 
> It would also be nice if once a person shares a DevCard during onboarding, they're taken to the profile edit page.
> 
> We don't want to leave them hanging after they share.
> 
> I'm not trying to nitpick, but I want to make sure users can see a clear path to usage.

Good point about the workspace destination. I think having a second button that goes to the workspaces page is good, it has a modal that pops up introducing workspaces in a video

bdougie commented:
> Good point about the workspace destination. I think having a second button that goes to the workspaces page is good, it has a modal that pops up introducing workspaces in a video

Second option should view your profile. If you logged in with tour GitHub, chances are you will want to see your profile.

Workspaces is a stretch today, and the scope for making that work from onboarding is something we should address just not in this PR

brandonroberts commented:
Updated the last step to preview your DevCard, and added a button to go to your profile

<img width="1274" alt="image" src="https://github.com/user-attachments/assets/84431e8f-fe6f-45fa-87d4-8c202fd099f6">
